### PR TITLE
Smuggler api tests are now passing 

### PR DESCRIPTION
**Pull request by fitzchak**
We currently import just the docs, and in one tx. 
We need to also import identities, (we do export them), and use the tx merger.

### CODE REVIEW

**Comment by redknightlois** [comment]
Just to let you know that Raven.Smuggler main project is not under `src/`. Dont know if it is fixed here already.

File: src/Raven.Server/Smuggler/SmugglerHandler.cs [hunk 1]
```diff
@@ -28,10 +32,103 @@ public Task PostExport()
...
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            //TODO: detect gzip or not based on query string param
+            using (var stream = new GZipStream(HttpContext.Request.Body, CompressionMode.Decompress))
+            {
+                var state = new JsonParserState();
+                using (var parser = new UnmanagedJsonParser(context, state, fileName))
+                {
+                    var buffer = context.GetParsingBuffer();
+                    int objectDepth = 0;
```
Comment (ayende) [inline src/Raven.Server/Smuggler/SmugglerHandler.cs:47]: Don't like this usage, it is very non obvious what is going on

File: src/Raven.Server/Smuggler/SmugglerHandler.cs [hunk 2]
```diff
@@ -28,10 +32,103 @@ public Task PostExport()
...
+                                        builder.FinalizeDocument();
+                                        using (var reader = builder.CreateReader())
+                                        using (var tx = context.OpenWriteTransaction())
+                                        {
+                                            BlittableJsonReaderObject metadata;
+                                            if (reader.TryGet(Constants.Metadata, out metadata) == false)
+                                                throw new InvalidOperationException("A document must have a metadata");
+                                            // We are using the @id here and not @key in order to be backward compatiable with old export files.
+                                            string key;
+                                            if (metadata.TryGet(Constants.MetadataDocId, out key) == false)
```
Comment (ayende) [inline src/Raven.Server/Smuggler/SmugglerHandler.cs]: Need to also handle newer versions here.
@id is the thing we use in the metadata, @key shouldn't be there

**Comment by redknightlois** [comment]
Check also if you have updated the CoreCLR tooling, cause the smuggler  project was linked against rc2 instead.

**Comment by ayende** [comment]
I think the smuggler project is dea

*Hibernating Rhinos Ltd  *

Oren Eini\* l CEO l *Mobile: + 972-52-548-6969

Office: +972-4-622-7811 *l *Fax: +972-153-4-622-7811

**Comment by fitzchak** [comment]
@redknightlois I created it long ago, before updating the tooling, so I removed it for now until we'll need it.

@ayende I think we're still going to have the exe, no?

File: src/Raven.Server/Smuggler/SmugglerHandler.cs (hunk 2 above)
Comment (fitzchak) [inline src/Raven.Server/Smuggler/SmugglerHandler.cs]: Constants.MetadataDocId is "@id", I'll change the variable name to also be id and not key

File: src/Raven.Server/Smuggler/SmugglerHandler.cs (hunk 1 above)
Comment (fitzchak) [inline src/Raven.Server/Smuggler/SmugglerHandler.cs:47]: This is WIP, sure it should be refactored

**Comment by ayende** [comment]
Yes, we are still going to have an exe, which just does a simple http call to the server

**Pull request merged**